docs/custom-filters.adoc

```diff
@@ -36,7 +49,106 @@ For common things like logging and metric facade APIs it is recommended to use t
 == Protocol filters
 
 A protocol filter is a `public` top-level, concrete class with a particular public constructor and which implements
-one or more protocol filter interfaces.
+one or more protocol filter interfaces. You can implement two distinct types of Custom Protocol Filter:
+
+- <<Specific Message Protocol Filters>>
+- <<Request/Response Protocol Filters>>
+
+Note that these types are mutually exclusive, for example a Filter is not allowed to implement RequestFilter and
+MetadataRequestFilter.
+
+You can also implement a convenience interface to deliver multiple Protocol Filters to the filter chain using <<Composite Filters>>
+
+=== Specific Message Protocol Filters
+
+A filter may wish to intercept specific types of Kafka messages. For example, intercept all Produce Requests, or
+intercept all Fetch Responses. To support this case Kroxylicious provides an interfaces for all request types and
+response types supported by Kafka (at the version of Kafka Kroxylicious depends on).
+
+So to intercept all Fetch Responses your class would implement
+{filter-api-javadoc}/io/kroxylicious/proxy/filter/FetchResponseFilter.html[FetchResponseFilter].
+To intercept all Produce Requests your Filter class would implement
+{filter-api-javadoc}/io/kroxylicious/proxy/filter/ProduceRequestFilter.html[ProduceRequestFilter].
+
+A Filter implementation is free to implement as many of these Specific Message interfaces as you like.
+
+For example:
+
+[source,java]
+----
+public class FetchRequestClientIdFilter implements FetchRequestFilter {
+
+    @Override
+    public void onFetchRequest(short apiVersion,
+                               RequestHeaderData header,
+                               FetchRequestData request,
+                               KrpcFilterContext context) {
+        header.setClientId("fetch-client!");
+        context.forwardRequest(header, request);
+    }
+}
+----
+
+=== Request/Response Protocol Filters
+
+A filter may wish to intercept every message being sent from the Client to Kafka or from Kafka to the Client. To do this
```

Comment:
Let's make sure we converge on a common parlance.

https://kroxylicious.io/kroxylicious/#_configuring_virtual_clusters:~:text=Kafka%20Cluster%20or%20Cluster says we will use the word "Kafka Cluster" or "Cluster" to refer to the Kafka Cluster.     You've just the word "Kafka" to refer to the Kafka Cluster, which I think is less clear.

```suggestion
A filter may wish to intercept every message being sent from the Client to the Cluster or from the Cluster to the Client. To do this
```

Reply:
Have updated all usages to `the Cluster`